tests/Stripe/ChargeTest.php

```diff
@@ -137,4 +137,41 @@ public function testCanMarkAsSafe()
         $this->assertInstanceOf("Stripe\\Charge", $resource);
         $this->assertSame($resource, $charge);
     }
+
+    public function testStatementDescriptorSanitization()
+    {
+        // Strings that are fine will pass.
+        $this->assertEquals(
+            'This is fine.',
+            Charge::sanitizeStatementDescriptor('This is fine.')
+        );
+
+        // Long strings are truncated.
+        $this->assertEquals(
+            'This string is longer ',
+            Charge::sanitizeStatementDescriptor('This string is longer than 22 characters.')
+        );
+
+        // Disallowed characters are stripped out.
+        $this->assertEquals(
+            '<b>"Disallowed\' bits</b>',
+            Charge::sanitizeStatementDescriptor('bDisallowed bits/b')
+        );
+
+        // A prefix is not applied if the descriptor is not numeric.
+        $this->assertEquals(
+            'Hello world.',
+            Charge::sanitizeStatementDescriptor('Hello world.', 'Example: ')
+        );
+
+        // A prefix is applied if the descriptor is numeric.
+        $this->assertEquals(
+            'Example: 123',
+            Charge::sanitizeStatementDescriptor('123', 'Example: ')
+        );
+
+        // A prefix must be supplied if the descriptor is numeric.
+        $this->expectsException(\InvalidArgumentException::class);
```

Comment:
Unfortunately, the library still needs to support PHP 5.4 so we cannot use `::class`.

Also, I don't think `expectsException` is a valid PHPUnit method -- afaik you have to use the [`@expectedException`](https://phpunit.de/manual/4.8/en/appendixes.annotations.html#appendixes.annotations.expectedException) annotation, which would require splitting this test into multiple methods.